stable/spark/docker_images/livy/Dockerfile

```diff
@@ -0,0 +1,137 @@
+# FROM ubuntu:16.04
+FROM java:openjdk-8-jdk
+MAINTAINER Dalitso Banda <dalitsohb@gmail.com>
+
+ENV LIVY_VERSION="git_master"
+ENV LIVY_COMMIT="02550f7919b7348b6a7270cf806e031670037b2f"
+ENV LOG_TAG="[LIVY_${LIVY_VERSION}]:" \
+    LIVY_HOME="/livy" \
+    LANG=en_US.UTF-8 \
+    LC_ALL=en_US.UTF-8
+
+RUN echo "$LOG_TAG Install essentials" && \
+    apt-get -y update && \
+    apt-get install -y locales && \
+    locale-gen $LANG && \
+    apt-get install -y git wget grep curl sed && \
+    apt-get autoclean &&  apt-get autoremove
+
+RUN echo "$LOG_TAG Getting maven" && \
+    wget http://www.eu.apache.org/dist/maven/maven-3/3.3.9/binaries/apache-maven-3.3.9-bin.tar.gz && \
+    tar -zxf apache-maven-3.3.9-bin.tar.gz -C /usr/local/ && \
+    ln -s /usr/local/apache-maven-3.3.9/bin/mvn /usr/local/bin/mvn 
+
+RUN curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py && \
+    python get-pip.py && \
+    rm get-pip.py && \
+    apt-get update && \
+    apt-get upgrade -y && \
+    apt-get install -y gcc python-dev libkrb5-dev && \
+    echo "deb http://deb.debian.org/debian stretch main" >> /etc/apt/sources.list && \
+    apt-get update && \
+    apt-get install -y g++ gcc-6 && \
+    # useful python packages
+    apt-get install -y python-dev libpython3-dev build-essential pkg-config gfortran && \
+    pip install -U pip setuptools wheel && \
+    # pip install -v pyspark && \
+    pip install numpy && \
+    pip install matplotlib && \
+    pip install pandas && \
+    pip install cloudpickle && \
+    pip install requests && \
+    pip install requests-kerberos && \
+    pip install flake8 && \
+    pip install flaky && \
+    pip install pytest && \
+    apt-get purge -y --auto-remove build-essential pkg-config gfortran libpython3-dev && \
+    apt-get autoremove && \
+    rm -rf /root/.cache
+
+RUN echo "$LOG_TAG Download and build Livy source" && \
+    git clone https://github.com/apache/incubator-livy.git ${LIVY_HOME}_src && \
+    cd ${LIVY_HOME}_src  && \
+    git checkout ${LIVY_COMMIT} && \
+    mvn package -DskipTests && \
+    ls ${LIVY_HOME}_src 
+
+# ENV LIVY_RELEASE_5_URL = "https://www-us.apache.org/dist/incubator/livy/0.5.0-incubating/livy-0.5.0-incubating-bin.zip"
+# RUN wget "https://www-us.apache.org/dist/incubator/livy/0.5.0-incubating/livy-0.5.0-incubating-bin.zip" 
+
+# RUN unzip livy-0.5.0-incubating-bin.zip && \
+#     mv livy-0.5.0-incubating-bin ${LIVY_HOME}
+
+
+
+RUN echo "$LOG_TAG install tini related packages" && \
+    apt-get install -y wget curl grep sed dpkg && \
+    TINI_VERSION=`curl https://github.com/krallin/tini/releases/latest | grep -o "/v.*\"" | sed 's:^..\(.*\).$:\1:'` && \
+    curl -L "https://github.com/krallin/tini/releases/download/v${TINI_VERSION}/tini_${TINI_VERSION}.deb" > tini.deb && \
+    dpkg -i tini.deb && \
+    rm tini.deb
+
+
```

Comment:
squish Run commands together to improve efficiency (removing files wont reduce size if its in a separate layer)